src/Indexing/Indexer.php

```diff
@@ -89,15 +109,25 @@ private function getIndexes($offset)
         if (empty($offset))
             return [];
 
-        // offset is an array of indexes already
-        if (!is_bool($offset[0]))
-            return $offset;
-
         // convert mask array to array of indexes
         foreach ($offset as $index => $value)
         {
-            if ($value)
-                $indexes[] = $index;
+            if ($value instanceof $this)
```

Comment:
I think it's bad comprehensibility.

Can you use switch-case operation? Like a
```php
 switch(true):{
  case (get_class($value) === this::class):
  // do smth...
  break;
  case ($value === true):
  // do smth...
  break;
  case ($value === false):
  break;
  default:
  // do smth
}
```

Reply:
Why do you think `get_class($value) === this::class` is better than `$value instanceof $this` ?

But anyway I'm going to refactor this code soon.

Reply:
This is the same, and the main idea of my comment is using switch-case statement with `true` argument. A little trick for better readable.

Reply:
that's good. I'll take it in consideration. Thanks!